libraries/chain/block_log.cpp

```diff
@@ -31,1490 +21,1619 @@ namespace eosio { namespace chain {
-         auto status = fseek(original_block_log.blk_in, start_of_blk_buffer_pos, SEEK_SET);
-         EOS_ASSERT( status == 0, block_log_exception, "original blocks.log seek failed" );
-         const auto num_read = fread(buf, read_size, 1, original_block_log.blk_in);
-         EOS_ASSERT( num_read == 1, block_log_exception, "original blocks.log read failed" );
-
-         // walk this memory section to adjust block position to match the adjusted location
-         // of the block start and store in the new index file
-         write_size = read_size;
-         while(original_pos >= start_of_blk_buffer_pos) {
-            const auto buffer_index = original_pos - start_of_blk_buffer_pos;
-            uint64_t pos_content = read_buffer<uint64_t>(buf + buffer_index);
-
-            if ( (pos_content - start_of_blk_buffer_pos) > 0 && (pos_content - start_of_blk_buffer_pos) < pos_size ) {
-               // avoid the whole 8 bytes that contains a blk pos being split by the buffer
-               write_size = read_size - (pos_content - start_of_blk_buffer_pos);
-            }
-            const auto start_of_this_block = pos_content;
-            pos_content = start_of_this_block - pos_delta;
-            write_buffer<uint64_t>(buf + buffer_index, &pos_content);
-            index.write(pos_content);
-            original_pos = start_of_this_block - pos_size;
-         }
-         new_block_file.seek(new_block_file_first_block_pos + to_write_remaining - write_size);
-         uint64_t offset = read_size - write_size;
-         new_block_file.write(buf+offset, write_size);
+      if (interval == 0) {
+         interval = std::max((log_bundle.log_index.num_blocks() + 7) >> 3, 1);
       }
+      uint32_t expected_block_num = log_bundle.log_data.first_block_num();
 
-      fclose(original_block_log.blk_in);
-      original_block_log.blk_in = nullptr;
-      new_block_file.flush();
-      new_block_file.close();
-
-      if (rename_input) {
-         fc::path old_log = output_dir / "old.log";
-         rename(original_block_log.block_file_name, old_log);
-         rename(new_block_filename, original_block_log.block_file_name);
-         fc::path old_ind = output_dir / "old.index";
-         rename(original_block_log.index_file_name, old_ind);
-         rename(new_index_filename, original_block_log.index_file_name);
+      for (uint32_t pos = 0; pos < log_bundle.log_index.num_blocks(); pos += interval, expected_block_num += interval) {
+         log_bundle.log_data.light_validate_block_entry_at(log_bundle.log_index.nth_block_position(pos),
+                                                           expected_block_num);
       }
+   }
 
-      return true;
+   std::pair<fc::path, fc::path> blocklog_files(const fc::path& dir, uint32_t start_block_num, uint32_t num_blocks) {
+      const int bufsize = 64;
+      char      buf[bufsize];
+      snprintf(buf, bufsize, "blocks-%d-%d.log", start_block_num, start_block_num + num_blocks - 1);
```

Comment:
These should be `%u` -- the arguments are unsigned ints.

Reply:
This is corrected now.